dataloader.go

```diff
@@ -20,18 +20,47 @@ import (
 // different access permissions and consider creating a new instance per
 // web request.
 type Interface interface {
-	Load(context.Context, interface{}) Thunk
-	LoadMany(context.Context, []interface{}) ThunkMany
-	Clear(context.Context, string) Interface
+	Load(context.Context, Keyer) Thunk
+	LoadMany(context.Context, KeyList) ThunkMany
+	Clear(context.Context, Keyer) Interface
 	ClearAll() Interface
-	Prime(ctx context.Context, key interface{}, value interface{}) Interface
+	Prime(ctx context.Context, key Keyer, value interface{}) Interface
+}
+
+// StringKey implements the Key interface for a string
+type StringKey string
+
+// Key is an identity method. Used to implement Key inteface
+func (k StringKey) Key() string { return string(k) }
+
+// String is an identity method. Used to implement Key Value
+func (k StringKey) Value() interface{} { return k }
+
+// Keyer is the interface that all keys need to implement
+type Keyer interface {
+	// Key returns a garunteed unique string that can be used to identify an object
+	Key() string
+	// Value returns the raw, underlaying value of the key
+	Value() interface{}
+}
+
+// KeyList is a slice of keys with some convience methods
+type KeyList []Keyer
+
+// Strings returns the list of strings. One for each "Key" in the list
+func (l KeyList) Strings() []string {
+	list := make([]string, len(l), len(l))
```

Comment:
We can omit the second `len(l)` call if we're setting both length and capacity to the same values.
```go
list := make([]string, len(l))
```